internal/services/applications/application_resource_test.go

```diff
@@ -594,6 +595,27 @@ func TestAccApplication_logo(t *testing.T) {
 	})
 }
 
+func TestAccApplication_password(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azuread_application", "test")
+	r := ApplicationResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.password(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("app_password.#").HasValue("1"),
+				check.That(data.ResourceName).Key("app_password.0.key_id").Exists(),
+				check.That(data.ResourceName).Key("app_password.0.value").Exists(),
+				check.That(data.ResourceName).Key("app_password.0.start_date").Exists(),
+				check.That(data.ResourceName).Key("app_password.0.end_date").Exists(),
+				check.That(data.ResourceName).Key("app_password.0.end_data_relative").Exists(),
+				check.That(data.ResourceName).Key("app_password.0.display_name").HasValue(data.ResourceName),
```

Comment:
These should be `password`, and we can remove the check for `end_date_relative`. We can also set a unique display name for the password for good measure.

```suggestion
				check.That(data.ResourceName).ExistsInAzure(r),
				check.That(data.ResourceName).Key("password.#").HasValue("1"),
				check.That(data.ResourceName).Key("password.0.key_id").Exists(),
				check.That(data.ResourceName).Key("password.0.value").Exists(),
				check.That(data.ResourceName).Key("password.0.start_date").Exists(),
				check.That(data.ResourceName).Key("password.0.end_date").Exists(),
				check.That(data.ResourceName).Key("password.0.display_name").HasValue(fmt.Sprintf("acctest-appPassword-%d", data.RandomInteger)),
```